src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -122,19 +174,19 @@ public class CosmosQueryEngine : IQueryEngine
                             new JProperty(QueryBuilder.PAGINATION_FIELD_NAME, jarray));
 
                         // This extra deserialize/serialization will be removed after moving to Newtonsoft from System.Text.Json
-                        return new Tuple<JsonDocument, IMetadata>(JsonDocument.Parse(res.ToString()), null);
+                        return res;
                     }
 
                     if (page.Count > 0)
                     {
-                        return new Tuple<JsonDocument, IMetadata>(JsonDocument.Parse(page.First().ToString()), null);
+                        return page.First();
```

Comment:
Instead of returning the tuple directly from help function we are returning the JObject and constructing the tuple at later stage. This is done in order to pass this helper function to func which then get executed with in Cache context. 
Tuple here was taking the JsonDocument as one of the input which was throwing the object dispose error while caching.

https://github.com/Azure/data-api-builder/blob/2ae5c0d7ab2ee03906d8fa8e70af40c086229e15/src/Core/Resolvers/CosmosQueryEngine.cs#L119C1-L120C1

Reply:
Right, but we don't need to store JsonElement either. I have it as JObject. We were converting the JObject to JsonDocument, which is no longer needed here.